test/e2e/test_system.py

```diff
@@ -417,3 +384,174 @@ def test_get_system_breadcrumbs_with_non_existent_id(test_client):
+
+
+def test_partial_update_system_invalid_parent_id(test_client):
+    """
+    Test updating a System's parent_id when the ID is invalid
+    """
+    # Create one to update
+    response = test_client.post("/v1/systems", json=SYSTEM_POST_A)
+    system = response.json()
+
+    # Update
+    response = test_client.patch(f"/v1/systems/{system['id']}", json={"parent_id": "invalid"})
+
+    assert response.status_code == 422
+    assert response.json()["detail"] == "The specified parent System ID does not exist"
+
+
+def test_partial_update_system_non_existent_parent_id(test_client):
+    """
+    Test updating a System's parent_id when the ID is non-existent
+    """
+    # Create one to update
+    response = test_client.post("/v1/systems", json=SYSTEM_POST_A)
+    system = response.json()
+
+    # Update
+    response = test_client.patch(f"/v1/systems/{system['id']}", json={"parent_id": str(ObjectId())})
+
+    assert response.status_code == 422
+    assert response.json()["detail"] == "The specified parent System ID does not exist"
+
+
+def test_partial_update_system_parent_id_duplicate_name(test_client):
+    """
+    Test updating a System's parent_id when the new parent has a child with a duplicate name
+    """
+    # Parent to move to
+    parent_system = test_client.post("/v1/systems", json=SYSTEM_POST_A).json()
+
+    # Two identical systems, one already with a parent the other without
+    test_client.post(
+        "/v1/systems", json={**SYSTEM_POST_B, "name": "Duplicate name", "parent_id": parent_system["id"]}
+    ).json()
+    system = test_client.post("/v1/systems", json={**SYSTEM_POST_B, "name": "Duplicate name"}).json()
+
+    # Change the parent of system to be the same as system1
+    response = test_client.patch(f"/v1/systems/{system['id']}", json={"parent_id": parent_system["id"]})
+
+    assert response.status_code == 409
+    assert response.json()["detail"] == "A System with the same name already exists within the parent System"
+
+
+def test_partial_update_system_name(test_client):
+    """
+    Test updating a System's name
+    """
+    _test_partial_update_system(test_client, {"name": "Updated name"}, {"code": "updated-name"})
+
+
+def test_partial_update_system_description(test_client):
+    """
+    Test updating a System's description
+    """
+    _test_partial_update_system(test_client, {"description": "Updated description"})
+
+
+def test_partial_update_system_location(test_client):
+    """
+    Test updating a System's location
+    """
+    _test_partial_update_system(test_client, {"location": "Updated location"})
+
+
+def test_partial_update_system_owner(test_client):
+    """
+    Test updating a System's owner
+    """
+    _test_partial_update_system(test_client, {"owner": "Updated owner"})
+
+
+def test_partial_update_system_importance(test_client):
+    """
+    Test updating a System's importance
+    """
+    _test_partial_update_system(test_client, {"importance": "high"})
```

Comment:
We probably don't need these because we don't have any logic that changes them or uses them for something else so we just trust that the dict merging etc works as expected. No harm in keeping them in but this was the only reason why I didn't test fields like this in the catalogue category or catalogue item entities. It gets long when the entity has many fields.

Reply:
Yeah I agree, although if I don't do this I would rather have one that just does all the fields at once instead, unlikely we would mess up the schema but if we did it would at least catch it via that and the others that only update name/parent id.